core/src/test/java/org/elasticsearch/index/query/TemplateQueryBuilderTests.java

```diff
@@ -18,21 +18,54 @@
  */
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.script.ScriptService.ScriptType;
 import org.elasticsearch.script.Template;
-import org.elasticsearch.test.ESTestCase;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
-/**
- * Test building and serialising a template search request.
- * */
-public class TemplateQueryBuilderTests extends ESTestCase {
+import static org.hamcrest.Matchers.is;
+
+public class TemplateQueryBuilderTests extends BaseQueryTestCase<TemplateQueryBuilder> {
+
+    /**
+     * The query type all template tests will be based on.
+     */
+    private static QueryBuilder<?> templateBase;
+
+    @BeforeClass
+    public static void setupClass() {
+        templateBase = RandomQueryBuilder.createQuery(getRandom());
+    }
+
+    @Override
+    protected boolean supportsBoostAndQueryName() {
+        return false;
+    }
+
+    @Override
+    protected TemplateQueryBuilder doCreateTestQueryBuilder() {
+        return new TemplateQueryBuilder(new Template(templateBase.toString()));
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(TemplateQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        if (!templateBase.equals(EmptyQueryBuilder.PROTOTYPE)) {
+            assertEquals(templateBase.toQuery(createShardContext()).getClass().getName(), query.getClass().getName());
+        }
+    }
```

Comment:
I checked out the branch, `testToQuery` fails consistently. You have to override `setFinalBoost` in `TemplateQueryBuilder` and make it a no-op like we did in `WrapperQueryBuilder`. Also this conditional in doAssertLuceneQuery shouldn't be needed anymore. Replace with a simple equality check between `query` and the result ot `templateBase#toQuery`. The problems you previously encountered around boost should be solved now.